### PR TITLE
Improve admin services discoverability.

### PR DESCRIPTION
**Pull request by johngian**
- Setup haproxy as reverse proxy to admin services
- Use an alternative port for jenkins (was port 80)
- Backend services
  - mesos
  - marathon
  - chronos
  - jenkins
  - consul
  - sensu

### CODE REVIEW

File: roles/admin/files/haproxy.ctmpl [hunk 1]
```diff
@@ -0,0 +1,58 @@
+{{- define "SERVICES_FQDN" -}}
+    services.mozilla.community
```
Comment (flamingspaz) [inline roles/admin/files/haproxy.ctmpl]: Let's keep our public jenkins (and other services in the future) under services.mozilla.community - we should use ops.mozilla.community instead.

File: roles/admin/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,120 @@
+---
```
Comment (flamingspaz) [inline roles/admin/tasks/main.yml:1]: All of these should include the "admin" tag so we can play through the entire role

**Review by flamingspaz — CHANGES_REQUESTED** [review]

**Comment by flamingspaz** [comment]
Also, change the consul.json.j2 for consul to reflect the new port

File: roles/admin/handlers/main.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+
+- name: restart haproxy
+  service:
+    name: haproxy
+    enabled: yes
```
Comment (flamingspaz) [inline roles/admin/handlers/main.yml:6]: Remove this line, we don't need to enable the service in the handler

File: roles/admin/handlers/main.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    state: restarted
+
+- name: restart consul-template
+  service:
+    name: consul-template
+    enabled: yes
```
Comment (flamingspaz) [inline roles/admin/handlers/main.yml:12]: Remove this line, we don't need to enable the service in the handler

File: roles/jenkins/templates/consul.json.j2 [hunk 5]
```diff
@@ -4,13 +4,13 @@
...
     "tags": ["jenkins", "admin", "{{ ec2_tag_env }}"],
-    "port": 80,
+    "port": 8081,
     "enableTagOverride": false,
     "checks": [
       {
       "id": "jenkins-{{ ec2_id }}",
       "name": "Jenkins on port 80",
-      "tcp": "{{ ansible_default_ipv4['address'] }}:80",
+      "http": "{{ ansible_default_ipv4['address'] }}:8081",
```
Comment (flamingspaz) [inline roles/jenkins/templates/consul.json.j2:13]: if we're using http checks the url needs to include `http://`. We could also use the /cc.xml endpoint so we're not pulling the entire html page each time

**Review by flamingspaz — CHANGES_REQUESTED** [review]

**Comment by johngian** [comment]
test this please

**Comment by johngian** [comment]
retest this please

**Comment by johngian** [comment]
retest this please

**Comment by johngian** [comment]
retest this please

**Comment by johngian** [comment]
retest this please

**Comment by johngian** [comment]
retest this please

File: roles/jenkins/templates/consul.json.j2 (hunk 5 above)
Comment (johngian) [reply roles/jenkins/templates/consul.json.j2:13]: It's not much of an overhead. Let's stick with the actual check of GETting the homepage.

**Pull request closed without merging**